pkg/nuctl/command/import.go

```diff
@@ -327,26 +317,65 @@ func (i *importProjectCommandeer) importProject(projectConfig *ProjectImportConf
 	}
 
 	functionImportErr := i.importFunctions(projectConfig.Functions, projectConfig.Project.Meta.Name)
-	functionEventImportErr := i.importFunctionEvents(projectConfig.FunctionEvents)
-
 	if functionImportErr != nil {
-		return errors.Wrap(functionImportErr, "Failed to import some functions")
+		i.rootCommandeer.loggerInstance.WarnWith("Unable to import all functions",
+			"functionImportErr", functionImportErr)
+
+		// return this error
+		err = functionImportErr
 	}
+
+	functionEventImportErr := i.importFunctionEvents(projectConfig.FunctionEvents)
 	if functionEventImportErr != nil {
-		return errors.Wrap(functionImportErr, "Failed to import some function events")
+		i.rootCommandeer.loggerInstance.WarnWith("Unable to import all function events",
+			"functionEventImportErr", functionEventImportErr)
+
+		// return this err only if not previously set
+		if err == nil {
+			err = functionEventImportErr
+		}
 	}
 
-	return nil
+	return err
 }
 
 func (i *importProjectCommandeer) importProjects(projectImportConfigs map[string]*ProjectImportConfig) error {
-	var g errgroup.Group
+	i.rootCommandeer.loggerInstance.DebugWith("Importing projects", "projects", projectImportConfigs)
 
+	// TODO: parallel this with errorGroup, mutex is required due to multi map writers
 	for _, projectConfig := range projectImportConfigs {
-		g.Go(func() error {
-			return i.importProject(projectConfig)
-		})
+		i.rootCommandeer.loggerInstance.DebugWith("Importing project",
+			"projectName", projectConfig.Project.Meta.Name)
+		if err := i.importProject(projectConfig); err != nil {
+			return errors.Wrap(err, "Failed to import project")
+		}
+	}
+	return nil
+}
+
+func (i *importProjectCommandeer) resolveProjectImportConfigs(projectBody []byte,
+	unmarshalFunc func(data []byte, v interface{}) error) (map[string]*ProjectImportConfig, error) {
+
+	// initialize
+	projectImportConfigs := map[string]*ProjectImportConfig{}
+
+	// try single
+	projectImportConfig := &ProjectImportConfig{}
+	if err := unmarshalFunc(projectBody, &projectImportConfig); err != nil {
+		return nil, errors.Wrap(err, "Failed to parse single project config")
```

Comment:
I suggest adding an explanation to this function or disambiguating this error. As at first glance, I couldn't understand how it worked.
Looking at this error, it looked to me as if - when you can't parse a single project, return an error.
When in reality, if you can't parse a single project you try multiple projects. However, if the parsing of the yaml/json fails in general because of a malformed json/yaml object, then any and all parsing will fail which should result in an error.